feedback-module/src/components/Module.js

```diff
@@ -119,12 +120,40 @@ function Module({ pageTitle, endpoint, dir }) {
     return checkedFields.some((field) => field.checked);
   };
 
+  //Check if valid email address
+  const invalidEmail = (email, required) => {
+    const re = /^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/;
+    let valid = re.test(email);
+    if (!required && email === "") {
+      valid = true;
+    }
+    return !valid;
+  };
+
+  const invalidPhone = (phone, required) => {
+    const reUS = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;
+
+    //Commenting this out because probably won't accept interntional phone numbers
+    //But keeping just in case:
+    // const reInternational = /^\+(?:[0-9] ?){6,14}[0-9]$/;
+    // let valid = reUS.test(phone) || reInternational.test(phone);
+    let valid = reUS.test(phone);
+    if (!required && phone === "") {
+      valid = true;
+    }
+    return !valid;
+  };
+
   //Checks if all the required fields have been completed - returns true if yes false if no
   const inputsValidated = () => {
     let validated = true;
     let questions = inputQuestions.map((question) => {
       if (question.required && question.answer === "") {
-        (question.error = true), (validated = false);
+        (validated = false), (question.error = true);
+      } else if (question.type === "email" && invalidEmail(question.answer)) {
```

Comment:
I noticed that the invalidEmail and invalidPhone functions both have a required parameter— should we be calling it with question.required as the 2nd argument?

Reply:
Wow yes I could have sworn I put that in. Weird that it is working without that